packages/components/src/Checkbox.tsx

```diff
@@ -76,7 +100,7 @@ export const Checkbox: ForwardRef<HTMLInputElement, CheckboxProps> =
           }}
         />
         <Box
-          as={CheckboxIcon}
+          as={(SVGProps) => CheckboxIcon({ isIndeterminate, ...SVGProps })}
```

Comment:
This won't be a performance problem here, but we usually wouldn't want to create new components like this.

Could we use the [`:indeterminate`](https://developer.mozilla.org/en-US/docs/Web/CSS/:indeterminate) pseudoclass for with general sibling combinator `~` to control what's displayed, just like we use `:checked` right now?

Reply:
I tried that first, but the attribute `indeterminate` on input cannot be set directly using HTML. And since none of the other components were performing side effects using hooks, I used this approach.

Reply:
I think we can go with the second option and use `<CheckboxIcon indeterminate={indeterminate}` as you said.

Reply:
Nice! I like this approach. Sorry for being so annoying here :)

Reply:
No worries,
Have made the changes.